### PR TITLE
[14.0][FIX] Fix failing tests

### PR DESCRIPTION
**Pull request by carmenbianca**

### CODE REVIEW

File: hotel_reservation/models/hotel_room.py [hunk 1]
```diff
@@ -182,29 +182,28 @@ def get_room_summary(self):  # noqa C901 (function is too complex)
...
                             [
                                 ("checkin", "<=", chk_date),
                                 ("checkout", ">=", chk_date),
-                                ("state", "=", "draft")
+                                ("state", "=", "draft"),
                             ]
                         )
                         for reservation in reservline_ids:
-                            for line in reservation.reservation_line.filtered(
+                            for _line in reservation.reservation_line.filtered(
```
Comment (robinkeunen) [inline hotel_reservation/models/hotel_room.py:193]: `_line` is not used. You could just name it `_`

File: hotel_reservation/tests/test_hotel_reservation.py [hunk 2]
```diff
@@ -24,7 +24,8 @@ def setUp(self):
         self.pricelist = self.env.ref("product.list0")
         self.floor = self.env.ref("hotel.hotel_floor_ground0")
         self.manager = self.env.ref("base.user_root")
-        cur_date = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        cur_date = datetime.now()
+        end_date = cur_date + timedelta(days=1)
```
Comment (robinkeunen) [inline hotel_reservation/tests/test_hotel_reservation.py:27]: `tomorrow`?

File: hotel_reservation/tests/test_hotel_reservation.py [hunk 3]
```diff
@@ -88,7 +88,7 @@ def setUp(self):
 
         self.hotel_room = self.hotel_room_obj.create(
             {
-                "product_id": self.room.product_id.id,
+                "product_id": self.room.product_id.copy().id,
```
Comment (robinkeunen) [inline hotel_reservation/tests/test_hotel_reservation.py:90]: I don't understand where `self.room.product_id` is set but anyway, I think it'd be better to create the product litteraly.

File: hotel_reservation/tests/test_hotel_reservation.py [hunk 4]
```diff
@@ -160,8 +159,7 @@ def test_reservation_reminder_24hrs(self):
         self.hotel_reserv.reservation_reminder_24hrs()
 
     def test_create_folio(self):
-        with self.assertRaises(ValidationError):
-            self.hotel_reserv.create_folio()
+        self.hotel_reserv.create_folio()
```
Comment (robinkeunen) [inline hotel_reservation/tests/test_hotel_reservation.py:162]: I guess the test was written to at least bring coverage if not testing. The `self.assertRaises` was put to pass the tests. Then you fixed the wrong behavior.

File: hotel_reservation/models/hotel_folio.py [hunk 5]
```diff
@@ -9,7 +9,7 @@ class HotelFolio(models.Model):
     _order = "reservation_id desc"
 
     reservation_id = fields.Many2one(
-        "hotel.reservation", "Reservation", ondelete="restrict"
+        "hotel.reservation", "Reservation", ondelete="set null"
```
Comment (robinkeunen) [inline hotel_reservation/models/hotel_folio.py:12]: Should'nt it be `cascade` ? If the folio is deleted, the linked reservation should be removed ? I often get confused with those.

**Review by robinkeunen — CHANGES_REQUESTED** [review]
I like the general idea, a few comments.

File: hotel_reservation/models/hotel_room.py (hunk 1 above)
Comment (carmenbianca) [reply hotel_reservation/models/hotel_room.py:193]: `_` is already used by Odoo's gettext, so can't use it.

File: hotel_reservation/tests/test_hotel_reservation.py (hunk 3 above)
Comment (carmenbianca) [reply hotel_reservation/tests/test_hotel_reservation.py:90]: A room is a product through delegation. This just copies the name, prices, etc etc etc.

File: hotel_reservation/models/hotel_folio.py (hunk 5 above)
Comment (carmenbianca) [reply hotel_reservation/models/hotel_folio.py:12]: You understood this the wrong way around :) It's the reservation that is being deleted, not the folio. `ondelete` determines what happens to the folio when that happens.

I'm not exactly sure what the correct choice here is.

Comment (robinkeunen) [reply hotel_reservation/models/hotel_folio.py:12]: Right, `set null` seems to be the most sensible.

**Pull request closed without merging**